Add task Solutions

Volodymyr-Bauzi opened the pull request:
[DEMO LINK](https://Volodymyr-Bauzi.github.io/react_todo-app-loading-todos/)

Volodymyr-Bauzi committed:
feat:
 - added loading todos when the `App` is loaded
 - added ability to hide the list and the footer if there are no todos
yet

 - added ability to close error notification with the `close` button
 - added ability to automatically hide the notification after 3 seconds> - added ability to hide the notification before any next request

 - added ability to filter todos by status `All` / `Active` / `Completed`

Volodymyr-Bauzi committed:
fix: linting error - A form label must be associated with a control (jsx-a11y/label-has-associated-control).

MariaAlefirenko reviewed (approved):
Good work!
Take the Filter out into components.
(⌒▽⌒)☆

Volodymyr-Bauzi committed:
feat:
- moved Footer to another component
- removed unnecesary comments

Volodymyr-Bauzi committed:
feat:
- created error block in different component
- added autofocus on main input
- add handleStatus handler
- add closeHandler to close error by clicking on it's 'X'

Volodymyr-Bauzi committed:
fix: minor linter problems

781calmstate commented on src/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,80 @@
+    if (titleRef.current) {
+      titleRef.current.focus();
+    }
+  }, [titleRef.current]);
```
the lint error telling the truth :)
